### PR TITLE
Common dao api

### PR DESCRIPTION
**Pull request by MyckolaPetrychko**

### CODE REVIEW

File: softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/ConversationEntity.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+import javax.persistence.*;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Created by pvm on 12.03.17.
+ */
+@Entity
+@Table(name = "Converstations")
```
Comment (Dominys) [inline softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/ConversationEntity.java]: @MyckolaPetrychko table names in DB are usually in lower case.

File: softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/ConversationEntity.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+public class ConversationEntity {
+
+    @Id
+    @GeneratedValue(generator = "uuid")
+    @GenericGenerator(name = "uuid", strategy = "uuid2")
+    private String uuid;
+
+    private Integer type;
```
Comment (Dominys) [inline softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/ConversationEntity.java]: @MyckolaPetrychko enum

File: softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/ConversationSettingsEntity.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
+package org.softgroup.common.dao.api.entities;
+
+import javax.persistence.*;
+
+/**
+ * Created by pvm on 12.03.17.
+ */
+@Entity
+@Table(name = "ConversationSettings")
```
Comment (Dominys) [inline softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/ConversationSettingsEntity.java]: @MyckolaPetrychko same

File: softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/ConversationSettingsEntity.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+public class ConversationSettingsEntity {
+    @Id
+    private Long id;
+
+    private String type;
```
Comment (Dominys) [inline softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/ConversationSettingsEntity.java]: @MyckolaPetrychko enum

**Comment by Dominys** [comment]
@MyckolaPetrychko All our entities will use String ID. Make some base Entity for this.

File: softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/ConversationEntity.java (hunk 2 above)
Comment (MyckolaPetrychko) [inline softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/ConversationEntity.java]: Enum for type?

Comment (MyckolaPetrychko) [inline softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/ConversationEntity.java]: I already know.

File: softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/DeviceEntity.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    @Column(name = "device_id")
+    private String deviceId;
+
+    @Column(name = "profile_id")
+    @ManyToOne(fetch = FetchType.LAZY)
+    private ProfileEntity profileId;
+
+    @Column(name = "last_confirmation")
+    private Timestamp lastConfirmation;
```
Comment (Dominys) [inline softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/DeviceEntity.java]: @MyckolaPetrychko Long

File: softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/DeviceTokenEntity.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+import javax.persistence.*;
+import java.security.Timestamp;
+
+/**
+ * Created by pvm on 14.03.17.
+ */
+@Entity
+@Table(name = "device_token")
+public class DeviceTokenEntity extends BaseEntity{
```
Comment (Dominys) [inline softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/entities/DeviceTokenEntity.java]: @MyckolaPetrychko Why you need this entity ??

File: softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/enums/ConversationType.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package org.softgroup.common.dao.api.enums;
+
+/**
+ * Created by pvm on 16.03.17.
+ */
+public enum ConversationType {
+    PESRONAL,
+    PUBLIC,
+    DEFAULT;
```
Comment (Dominys) [inline softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/enums/ConversationType.java]: @MyckolaPetrychko What is default ?

Comment (MyckolaPetrychko) [inline softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/enums/ConversationType.java]: It is PERSONAL and PUBLIC together.

Comment (Dominys) [inline softcourse/basis/common-dao/common-dao-api/src/main/java/org/softgroup/common/dao/api/enums/ConversationType.java]: @MyckolaPetrychko Conversation can be only personal and group.

**Pull request closed without merging**